components/registry-facade/pkg/registry/metrics.go

```diff
@@ -45,11 +45,12 @@ func (m *measuringRegistryRoundTripper) RoundTrip(req *http.Request) (*http.Resp
 
 // Metrics combine custom metrics exported by registry facade
 type metrics struct {
-	ManifestHist prometheus.Histogram
-	BlobCounter  prometheus.Counter
+	ManifestHist          prometheus.Histogram
+	BlobCounter           prometheus.Counter
+	BlobDownloadSpeedHist prometheus.Histogram
 }
 
-func newMetrics(reg prometheus.Registerer) (*metrics, error) {
+func newMetrics(reg prometheus.Registerer, upstream bool) (*metrics, error) {
```

Comment:
Could we have a comment explaining what is the use of the new argument? :grimacing: